WeatherService.Tests/BotsTest/WeatherObserverShould.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Moq;
+using System.Runtime.InteropServices;
+using WeatherMonitoringReportingService.Bots;
+using WeatherMonitoringReportingService.Data;
+using Xunit;
+namespace WeatherService.Tests.BotsTest
+{
+    public class WeatherObserverShould
+    {
+        private readonly Mock<IWeatherObserver> weatherObserver;
+        private readonly Mock<IWeatherStrategy> mockWeatherStrategy;
+        private readonly SunBot weatherBot;
+        private readonly WeatherData weatherData;
+
+        public WeatherObserverShould()
+        {
+            weatherObserver = new Mock<IWeatherObserver>();
+            mockWeatherStrategy = new Mock<IWeatherStrategy>();
+
+            weatherBot = new SunBot(mockWeatherStrategy.Object, new ConfigurationData());
+            weatherData = new WeatherData();
+        }
+
+        [Fact]
+        public void RegisterObserver()
+        {
+            weatherObserver.Object.RegisterObserver(weatherBot);
```

Comment:
What you're doing here is that you're testing the mock which doesn't provide any value, RegisterObserver will always be called. To break it to you: 

- You create a mock of `IWeatherObserver`.
- You called the `RegisterObserver` method on the mocked `IWeatherObserver.`
- You verified if the `RegisterObserver` method was called on the mocked `IWeatherObserver`.

to fix that you will need to test on the real implementation
something like this `private readonly IWeatherObserver weatherObserver;`